lxd/request/context.go

```diff
@@ -34,6 +28,32 @@ func GetCtxValue[T any](ctx context.Context, key CtxKey) (T, error) {
 
 // SetCtxValue sets the given value in the request context with the given key.
 func SetCtxValue(r *http.Request, key CtxKey, value any) {
-	rWithCtx := r.WithContext(context.WithValue(r.Context(), key, value))
-	*r = *rWithCtx
+	ctx := context.WithValue(r.Context(), key, value)
+	*r = *r.WithContext(ctx)
+}
+
+// IsRequestContext checks if the given context is a request context.
+// This is determined by checking the presence of the request information in the context.
+func IsRequestContext(ctx context.Context) bool {
+	if ctx == nil {
+		return false
+	}
+
+	_, ok := GetCtxInfo(ctx)
+	return ok
+}
+
+// GetCtxInfo gets the request information from the request context.
+func GetCtxInfo(reqContext context.Context) (*Info, bool) {
```

Comment:
It appears to be that in Go function names about contexts use the word "Context" rather than "Ctx", and its only the argument names that use `ctx`. So I think this should be `GetContextInfo`, `ContextInfo` or perhaps just `Info`.